server/src/main/java/org/candlepin/model/PoolCurator.java

```diff
@@ -682,13 +722,6 @@ public void batchDelete(List<Pool> pools) {
         for (Pool pool : pools) {
             currentSession().delete(pool);
         }
-        for (Pool pool : pools) {
-            // Maintain runtime consistency. The entitlements for the pool
-            // have been deleted on the database because delete is cascaded on
-            // Pool.entitlements relation
-            Hibernate.initialize(pool.getEntitlements());
```

Comment:
After talking with @barnabycourt  I think it would be better to instead of deleting this to check whether the collection is initialized and if it is then do the clear. This is very defensive, it will shield us from possible ConstraintViolationExceptions



```java
    /**
     * Batch deletes a list of pools.
     * @param pools
     */
    public void batchDelete(List<Pool> pools) {
        for (Pool pool : pools) {
            currentSession().delete(pool);
            // Maintain runtime consistency. The entitlements for the pool
            // have been deleted on the database because delete is cascaded on
            // Pool.entitlements relation
            if (Hibernate.isInitialized(pool.getEntitlements())) {
                pool.getEntitlements().clear();
            }
        }
    }
```